### PR TITLE
PHPCS: Fix errors in Jetpack root classes, part 2

### LINKED ISSUE #21951: PHPCS: Jetpack root classes
### Impacted plugin

Jetpack

### What

PHPCS

### How

```
	"projects/plugins/jetpack/class.frame-nonce-preview.php",
	"projects/plugins/jetpack/class.jetpack-admin.php",
	"projects/plugins/jetpack/class.jetpack-affiliate.php",
	"projects/plugins/jetpack/class.jetpack-autoupdate.php",
	"projects/plugins/jetpack/class.jetpack-cli.php",
	"projects/plugins/jetpack/class.jetpack-client-server.php",
	"projects/plugins/jetpack/class.jetpack-connection-banner.php",
	"projects/plugins/jetpack/class.jetpack-heartbeat.php",
	"projects/plugins/jetpack/class.jetpack-modules-list-table.php",
	"projects/plugins/jetpack/class.jetpack-network-sites-list-table.php",
	"projects/plugins/jetpack/class.jetpack-network.php",
	"projects/plugins/jetpack/class.jetpack-post-images.php",
	"projects/plugins/jetpack/class.json-api-endpoints.php",
	"projects/plugins/jetpack/class.json-api.php",
```

Note that this does not include class.jetpack.php itself.

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

* Assignments in conditionals.
* Multiple assignments.
* `json_encode` to `wp_json_encode`, `parse_url` to `wp_parse_url`,
  `urlencode` to `rawurlencode`.
* Whitelist some `print_r`, `trigger_error`, and `base64_encode`.
* Fix a missing `break`.
* Rename some local variables. Remove some unused variables.
* Parens in `( ++$i ) % 2`. 
* Flag an empty `elseif` case as ok. Logic probably works better that way.
* Flag some legacy properties and functions starting with `_` as ok. 
* Flag some error-ignoring as ok. 
* Whitelist a `serialize`/`unserialize` pair for now. `@todo` to check that
  nothing anywhere depends on it, since all the junk is public and it's in a
  potential base class.
* Yodas.
* Output escaping.
* Ignore a WordPress.DB.PreparedSQL.InterpolatedNotPrepared
* Nonce handling.
* Ignore juggling of global $post.
* WordPress.WP.EnqueuedResourceParameters
* Strict comparisons.
* And fix a random error I spotted in the previous PR.

#### Jetpack product discussion

Fixes #21951

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Make sure the things touched still work. Particularly the added nonce handling in the Jetpack network sites page.

### CODE REVIEW

File: projects/plugins/jetpack/class.jetpack-admin.php [hunk 1]
```diff
@@ -128,7 +130,7 @@ public function akismet_logo_replacement_styles() {
 	 * @return int Indicating the relative ordering of module1 and module2.
 	 */
 	public static function sort_requires_connection_last( $module1, $module2 ) {
-		if ( $module1['requires_connection'] == $module2['requires_connection'] ) {
+		if ( (bool) $module1['requires_connection'] === (bool) $module2['requires_connection'] ) {
```
Comment (kraftbj) [inline projects/plugins/jetpack/class.jetpack-admin.php:133]: Out of curiosity, does this do anything different than the fuzzy equiv?

File: projects/plugins/jetpack/class.jetpack-modules-list-table.php [hunk 2]
```diff
@@ -319,12 +326,13 @@ public function column_cb( $item ) {
...
 	public function column_icon() {
-		$badge_text = $free_text = '';
+		$badge_text = '';
+		$free_text  = '';
 		ob_start();
 		?>
 		<a href="#TB_inline?width=600&height=550&inlineId=more-info-module-settings-modal" class="thickbox">
 			<div class="module-image">
-				<p><span class="module-image-badge"><?php echo $badge_text; ?></span><span class="module-image-free" style="display: none"><?php echo $free_text; ?></span></p>
+				<p><span class="module-image-badge"><?php echo esc_html( $badge_text ); ?></span><span class="module-image-free" style="display: none"><?php echo esc_html( $free_text ); ?></span></p>
```
Comment (kraftbj) [inline projects/plugins/jetpack/class.jetpack-modules-list-table.php:335]: I don't see a point to either of these variables. They're both empty and echo without modification. Do you see something I may be missing?

File: projects/plugins/jetpack/class.jetpack-admin.php (hunk 1 above)
Comment (anomiex) [reply projects/plugins/jetpack/class.jetpack-admin.php:133]: Depends on what the values are set to. If they're already always booleans, or 0/1, or something like that, then no. It does if for example `$module1['requires_connection']` is 1 while `$module2['requires_connection']` is 2 or something like that.

The code just below seems to be expecting it to be boolean, so this seemed like the expected behavior to me. Also grepping through the code it looks like things that set it are setting it to booleans.

**Review by kraftbj — APPROVED** [review]

**Comment by anomiex** [comment]
r238449-wpcom

**Pull request merged**